apps/src/blockly/addons/cdoUtils.js

```diff
@@ -214,17 +214,19 @@ export function registerCustomProcedureBlocks() {
 /**
  * Partitions blocks of the specified types to the front of the list.
  *
- * @param {Element[]} blockElements - An array of block elements to be partitioned.
- * @param {string[]} prioritizedBlockTypes - An array of strings representing block types.
- *    These types are moved to the front of the list while otherwise maintaining order.
- * @returns {Element[]} A new array of block elements partitioned based on their types.
+ * @param {Element[]|Object[]} blocks - An array of block elements or JSON blocks to be partitioned.
+ * @param {Object} [options] - An object containing partitioning options.
+ * @param {string[]} [options.prioritizedBlockTypes] - An array of strings representing block types to move to the front.
+ * @param {boolean} [options.isJson] - A flag indicating whether the blocks are JSON blocks (vs. block elements).
+ * @returns {Element[]|Object[]} A new array of block elements or JSON blocks partitioned based on their types.
  */
-export function partitionBlocksByType(blockElements, prioritizedBlockTypes) {
+export function partitionBlocksByType(blocks = [], options = {}) {
```

Comment:
nit: is there a reason you went with an options object rather than having separate parameters for each option? I think separate parameters is more readable.

Reply:
I can switch back to separate parameters! Once I get more than 2 or 3 parameters, I start to look for ways to have fewer than that, but I agree it's probably fine here.